examples/kubernetes/connectivity-check/echo-c.yaml

```diff
@@ -0,0 +1,50 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: echo-c
+spec:
+  type: ClusterIP
+  ports:
+  - port: 80
+  selector:
+    name: echo-c
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: echo-c
```

Comment:
nit: conformance test is passed right now, but we might need to add steps to dump details and logs in github action. 
This https://github.com/cilium/cilium/blob/master/.github/workflows/smoke-test.yaml#L153 is my naive attempt, and little verbose.

Reply:
@sayboras is there something we can do from the YAMLs to assist this?

My typical debugging involves basically `kubectl describe [...]` which hopefully tells me which pod + which container is failing, then from there I can try to reason about why that specific check will have failed.

Reply:
Thanks for your offer, but seems unlikely we can do much.

I would love to get to bottom of this https://github.com/cilium/cilium/issues/12279, but it's not failing for my self hosted github action. Current logs we have from github actions didn't help much.

Reply:
I've added a new commit on top that hopes to improve the debugging, and also covers dumping logs for all of the services (by iterating the cue definitions rather than listing them one-by-one).